interpreter/cling/lib/Interpreter/NullDerefProtectionTransformer.cpp

```diff
@@ -122,79 +127,42 @@ namespace cling {
           stmts.append(nc.getStmts().begin(), nc.getStmts().end());
       }
 
-      llvm::ArrayRef<Stmt*> stmtsRef(stmts.data(), stmts.size());
-      CompoundStmt* newCS = new (C) CompoundStmt(C, stmtsRef,
-                                                 CS->getLBracLoc(),
-                                                 CS->getRBracLoc());
-      return NodeContext(newCS);
-    }
-
-    NodeContext VisitIfStmt(IfStmt* If) {
-      // check the condition
-      NodeContext cond = Visit(If->getCond());
-      if (!cond.isSingleStmt())
-        If->setCond((clang::Expr*)cond.getStmt());
-
-      // No default constructor..
-      NodeContext result(If);
-      return result;
-    }
-
-    NodeContext VisitCastExpr(CastExpr* CE) {
-      NodeContext castExpr = Visit(CE->getSubExpr());
-      if (castExpr.isSingleStmt())
-        CE->setSubExpr((clang::Expr*)castExpr.getStmt());
+      if (!stmts.empty()) {
+        llvm::ArrayRef<Stmt*> stmtsRef(stmts.data(), stmts.size());
+        CS->setStmts(C, const_cast<Stmt**>(&stmtsRef.front()), stmts.size());
```

Comment:
According to the doc, `setStmts` takes an `ArrayRef` (in the clang master...). Can we just pass in `stmtsRef` itself, or do we really need to pass in the `Stmt**`?